File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    r, c = len(arr), len(arr[0])
    if r > c:
        for i in range(r):
            arr[i].extend([0]*(r-c))
    elif r < c:
        for _ in range(c-r):
            arr.append([0]*(c))
    return arr

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.rb
def solution(arr)
  rows = arr.length
  cols = arr[0].length

  if rows > cols
    arr.each do |row|
      (rows - cols).times { row << 0 }
    end
  elsif cols > rows
    (cols - rows).times { arr << Array.new(cols, 0) }
  end

  arr
end
